src/main/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -294,6 +301,99 @@ public String md5() {
         }
     }
 
+
+    /**
+     * Given a fasta filename, return the name of the corresponding dictionary file.
+     */
+    public static IOPath getFastaDictionaryFileName(IOPath fastaFile) {
+        final String fastaName = fastaFile.getURIString();
+        int lastDot = fastaName.lastIndexOf('.');
+        return new HtsPath(fastaName.substring(0, lastDot) + FileExtensions.DICT);
+    }
+
+    /**
+     * Given a fasta dictionary file, returns its sequence dictionary
+     *
+     * @param fastaDictionary fasta dictionary file
+     * @return the SAMSequenceDictionary from fastaDictionaryFile
+     */
+    public static SAMSequenceDictionary loadSAMSequenceDictionary( final IOPath fastaDictionary ) {
+        try ( final InputStream fastaDictionaryStream = fastaDictionary.getInputStream() ) {
+            return loadSAMSequenceDictionary(fastaDictionaryStream);
+        }
+        catch ( IOException e ) {
+            throw new RuntimeIOException("Error loading fasta dictionary file " + fastaDictionary, e);
+        }
+    }
+
+    /**
+     * Given an InputStream connected to a fasta dictionary, returns its sequence dictionary
+     *
+     * Note: does not close the InputStream it's passed
+     *
+     * @param fastaDictionaryStream InputStream connected to a fasta dictionary
+     * @return the SAMSequenceDictionary from the fastaDictionaryStream
+     */
+    public static SAMSequenceDictionary loadSAMSequenceDictionary( final InputStream fastaDictionaryStream ) {
+        // Don't close the reader when we're done, since we don't want to close the client's InputStream for them
+        final BufferedLineReader reader = new BufferedLineReader(fastaDictionaryStream);
+
+        final SAMTextHeaderCodec codec = new SAMTextHeaderCodec();
+        final SAMFileHeader header = codec.decode(reader, fastaDictionaryStream.toString());
+
+        // Make sure we have a valid sequence dictionary before continuing:
+        if (header.getSequenceDictionary() == null || header.getSequenceDictionary().isEmpty()) {
+            throw new RuntimeException (
+                    "Could not read sequence dictionary from given fasta stream " +
+                            fastaDictionaryStream
+            );
+        }
+
+        return header.getSequenceDictionary();
+    }
+
+    final void requireMD5sOrThrow(final String contextMessage) {
+        final List<SAMSequenceRecord> missingMD5s = getSequencesWithMissingMD5s();
+        if (!missingMD5s.isEmpty()) {
+            throw new RuntimeException(SAMSequenceDictionary.createFormattedMD5Message(contextMessage, missingMD5s));
+        }
+    }
+
+    /**
+     * @return all sequences in this dictionary that do not have an MD5 attribute
+     */
+    List<SAMSequenceRecord> getSequencesWithMissingMD5s() {
```

Comment:
These seem like very specific methods.  It's a bit funny to have them here as package private.  I wonder if it makes more sense to move them to the callsite.